src/App.tsx

```diff
@@ -1,27 +1,99 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import { Box, Button, Typography } from '@mui/material';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isActiveButtonLag, setIsActiveButtonLag] = useState(false);
+  const [isActiveButtonL5fg, setIsActiveButtonL5fg] = useState(false);
+  const [isActiveButtonLrg, setIsActiveButtonLrg] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const loadAllGoods = async () => {
+    const allGoods = await getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    setGoods(allGoods);
+    setIsActiveButtonLag(true);
+    setIsActiveButtonL5fg(false);
+    setIsActiveButtonLrg(false);
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const load5First = async () => {
+    const fiveFirst = await get5First();
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    setGoods(fiveFirst);
+    setIsActiveButtonLag(false);
+    setIsActiveButtonL5fg(true);
+    setIsActiveButtonLrg(false);
+  };
+
+  const loadRedGoods = async () => {
+    const redGoods = await getRedGoods();
+
+    setGoods(redGoods);
+    setIsActiveButtonLag(false);
+    setIsActiveButtonL5fg(false);
+    setIsActiveButtonLrg(true);
+  };
```

Comment:
If you can merge this 3 func in one, just with case and pass a func as arg

Reply:
fixed